Framework/Core/src/DeviceSpecHelpers.cxx

```diff
@@ -626,15 +626,18 @@ void DeviceSpecHelpers::processInEdgeActions(std::vector<DeviceSpec>& devices,
 // Construct the list of actual devices we want, given a workflow.
 //
 // FIXME: make start port configurable?
-void DeviceSpecHelpers::dataProcessorSpecs2DeviceSpecs(WorkflowSpec const& workflow,
-                                                       std::vector<ChannelConfigurationPolicy> const& channelPolicies,
-                                                       std::vector<CompletionPolicy> const& completionPolicies,
-                                                       std::vector<DispatchPolicy> const& dispatchPolicies,
+
+void DeviceSpecHelpers::dataProcessorSpecs2DeviceSpecs(const WorkflowSpec& workflow,
+                                                       DriverInfo& driverInfo,
                                                        std::vector<DeviceSpec>& devices,
-                                                       ResourceManager& resourceManager,
-                                                       std::string const& uniqueWorkflowId)
+                                                       ResourceManager& resourceManager)
 {
```

Comment:
Why do you need to expose the whole `DriverInfo` to this function?

Reply:
to avoid passing parameters one by one, specially when someone (in this case it was me) add new variable to DriverInfo

Reply:
I guess my point is, do you think we really need to expose the struct as whole, always? `DriverInfo` is a catch all struct for whatever pertains to the driver, so I would expect people to pick what they need. In this particular case why does `dataProcessorSpecs2DeviceSpecs` need to know about the `completionPolicy`?

Reply:
Well I just thought it would be easier to add more options using DriverInfo object without changing prototype of method dataProcessorSpecs2DeviceSpecs. But now i see your point, so i will change it.